### DISCUSSION TITLE
Can I claim a username for an organization?

### DISCUSSION THREAD

**Discussion by torralbaa** [community/community#23468] [New to GitHub]
<p>Hello,</p>
<p>If I want my organization to be called <em>X</em> and <em>X</em> is occupied by an account created 5 years or more ago that has 0 activities since its creation, can I claim <em>X</em> for my organization?</p>
<p>If so, how do I do it?</p>
<p>Thanks and sorry for my English.</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/alvarito050506">@alvarito050506</a>,</p>
<p>Thanks for being here! To check whether the name is available under our “name squatting policy” you’ll have to write into private support. If you haven’t already, please contact us at <a href="https://github.com/contact" rel="nofollow noopener">https://github.com/contact</a> with the info so we can look into that further. </p>

**Comment by torralbaa** [comment]
<p>Hi <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a>,</p>
<p>Thank you very much for your response, I will contact support.</p>
<p>Regards.</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hey <a class="mention" href="/alvarito050506">@alvarito050506</a>,</p>
<p>I’ve just learned of a change to our policy: </p>
<ul>
<li>Usernames at GitHub are offered on a first-come-first-served basis, and are intended for immediate use. If a name might be available for you to use, you will find that out during the process of <a href="https://help.github.com/en/articles/changing-your-github-username" rel="nofollow noopener">changing your GitHub username</a>.</li>
</ul>
<p>I hope this helps!</p>

**Comment by torralbaa** [comment]
<p>Hi, <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a>,</p>
<p>As far as I can see, this process is only valid for users and not for organizations.</p>
<p>Anyway thanks for the support!</p>

**Comment by AndreaGriffiths11** [comment]
<p>Got it! Thank you so much for the clarification.</p>

**Comment by agabor** [comment]
<p>The accepted answer does not seem to be true anymore. If I try to contact support with such issue, I get the following automated message:</p>
<blockquote>
<p><strong>GitHub no longer accepts requests to release dormant usernames .</strong></p>
<p>It appears that you are looking to contact us regarding releasing a dormant or inactive username. We no longer review individual requests to release dormant or inactive usernames.</p>
</blockquote>
<p>In my case there is a dormant account that happens to have the same name as my company, which did not have any activity for ten years. Is there really nothing to be done?</p>

**Reply by MartinThoma** [reply]
Did you find a solution?

**Reply by emmahsax** [reply]
But also just remember that although it may look like they don't have any activity, they may have private repositories they're actively committing to and working on.

**Comment by tuves** [comment]
<p>Hi <a class="mention" href="/agabor">@agabor</a>,</p>
<p>That is correct, we are no longer accepting requests to release dormant usernames. We are not able to accommodate this request.</p>
<p>If you hold a trademark to this name and believe there is a clear case of confusion between the account and your trademark, please review our <a href="https://docs.github.com/articles/github-trademark-policy/">Trademark Policy</a> to determine if you should submit a trademark policy violation report.</p>
<p>Thank you</p>

**Comment by SmashingQuasar** [comment]
@tuves Reacting to this post to open a discussion about the state of Github regarding organisations.
With time, there is an increasingly higher number of organisations.
Thousands of those organisations haven't created a repository or done any action for many years. Those organisations not only pollute Github as a whole but prevent active developers to dive in Github without spending hours if not days for an organisation name that is available and is it not a random string of characters.
Whilst not accommodating individual requests is understandable, letting dormant organisations exist for over 10 years is not. Github should start thinking about what is going to happen to those inactive organisations or in 10 years we will reach a point where it will be impossible to create any new organisation with a name that does not look like a UUID.
This is a common problem that is not exclusive to Github obviously, but here it is necessary for Github to face the reality of the issue and take global actions regarding this. It would also benefit Github to not have to store tens of thousands of dummy organisations.
I'm not even talking about inactive organisations, I'm talking about organisations that are entirely empty and have been for years without a single activity registered.

**Reply by ettaboyle** [reply]
@SmashingQuasar 

Please note, contacting any member of GitHub Staff via unsolicited mentions or pings is strongly discouraged and may be considered a violation of our prohibition against harassment. Please review our [Code of Conduct](https://docs.github.com/en/site-policy/github-terms/github-community-code-of-conduct) for more information.

Let's work together to keep GitHub Community a place where people feel safe to participate by being respectful of them and their time.

Any repeat and future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.

**Reply by SmashingQuasar** [reply]
Hey,
Thanks for your message.

Please not that you are refusing to enforce your own [terms of service](https://docs.github.com/en/site-policy/other-site-policies/github-username-policy#name-squatting-policy):
![image](https://github.com/community/community/assets/11139591/897ccc5a-64f5-456a-a9f2-11a20d16cd3e)

Considering those terms of service are legally binding for users, unlike your forum code of conduct, I will kindly ask you to revise your attitude towards this issue. Regarding your message, I will also mention your [own Forum Code of Conduct](https://docs.github.com/en/site-policy/github-terms/github-community-code-of-conduct#what-is-not-allowed):
![image](https://github.com/community/community/assets/11139591/94f2c742-0117-430b-957e-4991d24c721c)
As mentioned here, Github is a professional setting, which means professional conduct is expected, especially coming from Github. Refusing to enforce your own Terms of Service is highly unprofessional. 
This behaviour from Github is causing harm to entrepreneurs and companies trying to engage in fair use of Github. 

Thanks for understanding.

**Comment by algoflows** [comment]
Unbelievable, would have thought Github would have a process for this, I have also ran into a situation that two of my new org projects names can't be used because two users have squatted the names and are not active for years! I will try and pick this up with Github support and see what can be done. I would find it unbelievably short sited if GH don't have a process for getting org names for active organisation.

**Reply by spandres** [reply]
Same problem here. Should we start building a new RepoHub?

**Comment by peet86** [comment]
Same issue here.. in my case the desired name belongs to an organization which has 0 members, 0 repos, no activity, nothing.. I guess it's stuck forever..

**Comment by 8BallBomBom** [comment]
Why not just give organisations a different url?
It would annoy plenty but would essentially help to alleviate the issue.

Existing organisations would have to be redirected or give up their existing urls but it would at least give separation between users and organisations.

https://github.com/org/yourorganisation

Just an example.

**Reply by AxZupancic** [reply]
I know this is a couple years old now, but we are currently dealing with this too. Why on earth Github doesn't separate organization URLs from personal URLs is beyond me. GitLab here I come!

**Reply by cptcontainers** [reply]
My suggestion would be github.com/u/youruser for usernames and github.com/o/yourorganisation for organisations. Of course could also keep users with existing urls, but I actually think it's helpful that the url clearly distinguishes users and orgs... and it also leaves it open for github to easily introduce other categories of users in future if they need to.

**Comment by landsman** [comment]
No longer supported.

https://docs.github.com/en/site-policy/other-site-policies/github-username-policy#name-squatting-policy
https://support.github.com/contact/account

![Screenshot 2023-08-08 at 10 40 13](https://github.com/community/community/assets/3184228/477d85d5-de63-4531-84b5-2e460016bdac)

**Reply by miguelpeniche** [reply]
You need to see previous posts before posting my friend. Just saying.

**Comment by torralbaa** [comment]
Since name squatting is no longer a possibility, I think this should be closed as outdated.

Not that the situation is ideal, but there's just nothing to do about it.

**Reply by SmashingQuasar** [reply]
Name squatting is prohibited by the ToS, but this topic makes it clear you have no way of actually requesting any action against name squatting, therefore it is a rule with no enforcement, which is called a recommendation. Until Github lives up to their own ToS, this topic stays relevant.

**Reply by torralbaa** [reply]
Yes, but I wouldn't expect GH to listen to its users anymore, after all these atrocious changes they've implemented and not rolled back even after the more-or-less generalized backlash.

**Reply by cptcontainers** [reply]
This definitely seems like something that would make sense to revisit and I would keep this thread open. I agree that currently the name squatting policy in the Github ToS isn't being effectively enforced and there is quickly going to be a problem with lack of usable user and org names if something isn't done.

**Comment by cptcontainers** [comment]
This definitely seems like something that would make sense to revisit. Maybe even implementing some automated name squatting check if people request it... I understand Github staff don't want to be inundated with requests from people complaining that their preferred name was taken.

Additionally separating users and orgs definitely makes sense in my opinion... all that would be needed is to make it github.com/u/username and github.com/o/orgname and you could have a transitional period of a year where the existing url redirects to the new url and informs the visitor that url with out /u/ or /o/ will stop working and to update any bookmarks or links etc.

**Reply by 8BallBomBom** [reply]
Totally agree, would at least be better than what we have now...

**Comment by graybeal** [comment]
@AndreaGriffiths11 This question is marked as 'Answered by (you)', but (no offense) the marked answer is not correct, and the question/issue has never been fully addressed. 

The original problem continues to be a major obstacle to many GitHub users (obviously per Google search), as well as looking bad in PR terms. Can you please try to pursue some action to move these needs forward?

**Reply by ettaboyle** [reply]
Hi @graybeal 

Please note, contacting any member of GitHub Staff via unsolicited mentions or pings is strongly discouraged and may be considered a violation of our prohibition against harassment. Please review our [Code of Conduct](https://docs.github.com/en/site-policy/github-terms/github-community-code-of-conduct) for more information.

Let's work together to keep GitHub Community a place where people feel safe to participate by being respectful of them and their time.

Any repeat and future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.

**Reply by 4po** [reply]
que

**Reply by uptudev** [reply]
Please note that replying to someone to let them know that an accepted answer is outdated does not constitute harassment. Given the strict legal definition of the term, I would reconsider using it so freely and accusatorily in a professional setting to users seeking genuine support from public-facing staff members, as under the ***What is not Allowed*** section of said [Code of Conduct](https://docs.github.com/en/site-policy/github-terms/github-community-code-of-conduct#what-is-not-allowed), you will find the following:

> * Other conduct which could reasonably be considered inappropriate in a professional setting. GitHub Community is a professional space and should be treated as such.

Note that many of your users may find it **extremely unprofessional** to accuse them of harassment—and threaten to take action on their accounts—for using functionality built into the site to reply to a support conversation in a cordial manner, one which is conventional to any other support board.

Thank you for your understanding.

**Reply by SmashingQuasar** [reply]
@uptudev ettaboyle's comment is indeed absolutely unacceptable.

**Reply by uptudev** [reply]
@SmashingQuasar 

Like I'd understand it being considered "harassment" if somebody tagged a staff member who hadn't been in the conversation prior (or tagged them in a non-support thread), but replies to a **support** thread do not constitute harassment in any way; when you say something in a support thread as someone who works in support, you don't just get to ban people for replying to you. Note that Andrea even says "Thanks for the ping" in her reply because that's normal behaviour.

If they addressed ToS violations with the same zeal that they use to make up false CoC violations (breaking the CoC themselves in the process), this thread wouldn't even exist. Just unsubscribed, as I do not want my hard-earned money paying the wages of someone capable of such blatant disrespect while claiming that they're keeping "GitHub Community a place where people feel safe to participate by being ***respectful*** of them and their time".

**Comment by AndreaGriffiths11** [comment]
Thanks for the ping @graybeal, I'm afraid the guidance from our Support team has not changed. 
Keep in mind that not all activity on GitHub is publicly visible; accounts with no visible activity may be in active use.

**Reply by SmashingQuasar** [reply]
This does not address copyright infrigement. Github needs to have a policy to address these issues that does not systematically mean engaging into a cease and desist action.

**Reply by michaelfreund** [reply]
Dear @AndreaGriffiths11

I understand the rationale behind this decision, recognizing the significant effort involved in managing such requests. However, this policy can be to the detriment of many honest members of the open-source community.

For those of us deeply embedded in the open-source world, clear and concise usernames are not just identifiers; they are a vital part of our identity and brand within the community. These names contribute significantly to recognition and visibility, elements crucial for collaboration and the spread of ideas. Unfortunately, the current policy inadvertently favors individuals who reserve names without contributing activity, hindering active community members from obtaining meaningful identifiers that reflect their projects or initiatives.

This issue is not just theoretical for me; it's a current challenge. Our start-up faces this very problem. The ideal GitHub username for our project is occupied by an inactive account, forcing us to opt for a less intuitive alternative. This situation is frustrating, not only because it complicates our branding and community building efforts but also because it underscores a broader issue within the platform's approach to community engagement and support.

While I acknowledge that some users may prefer private repositories and their lack of public activity does not necessarily equate to inactivity, I believe there is a middle ground to be found. GitHub could explore innovative solutions to this issue, such as implementing a policy for reclaiming genuinely inactive usernames without compromising the privacy or security of users who choose to work privately. This could involve criteria such as the duration of inactivity, the total absence of contributions, or a requirement for account holders to confirm their continued interest in retaining their username periodically.

In conclusion, while I commend GitHub for its efforts to streamline operations and protect user interests, I strongly believe that revisiting the Username Policy with a view towards supporting active and contributing community members could significantly enhance the platform's value for the open-source ecosystem. A more dynamic and community-focused approach to usernames could foster an environment where innovation and collaboration are not hindered by administrative policies but are instead celebrated and encouraged.

Best
Michael

**Reply by miguelpeniche** [reply]
Hey @AndreaGriffiths11, you guys keep saying not all activity on GitHub is publicly visible and accounts with no visible activity **may be in active use**. I think I speak from everyone saying that we totally support that, those accounts have their right to continue as they were claimed first. The real debate is what happens to accounts that **may be in inactive use** for years? Is clear that we are talking about people that just created an organization and never used it or just created their account and never used it as well. What is the policy on that? Are those usernames gone for good? A new discussion should be done in Github HQ.

**Reply by miguelpeniche** [reply]
By the way, Heroku have a fair policy for their accounts:

> Heroku automatically deletes inactive accounts. An account is inactive if its owner hasn’t logged into the Dashboard, CLI, or API in the last 365 days, and no paid resources were used with Heroku in the last 6 months. To keep your account active, you must log into your account.

Source: https://devcenter.heroku.com/articles/account-management#auto-deletion-of-inactive-accounts

<img width="790" alt="heroku" src="https://github.com/community/community/assets/3253845/f30d822f-6f0d-4f2f-9e4d-91db9829e5e5">

**Reply by ettaboyle** [reply]
Hi there,

@michaelfreund & @miguelpeniche Please note, contacting any member of GitHub Staff via unsolicited mentions or pings is strongly discouraged and may be considered a violation of our prohibition against harassment. Please review our [Code of Conduct](https://docs.github.com/en/site-policy/github-terms/github-community-code-of-conduct) for more information.

Let's work together to keep GitHub Community a place where people feel safe to participate by being respectful of them and their time.

Any repeat and future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.

**Reply by michaelfreund** [reply]
Hi there,

First, I want to acknowledge the importance of the community guidelines highlighted in response to my previous post, specifically regarding direct mentions of GitHub staff. I understand these rules help maintain a respectful environment, and I will ensure to follow these guidelines in the future.

I would like to revisit the topic that I believe is crucial for many of us here: GitHub's username policy. This issue significantly impacts our ability to maintain a clear and effective presence on the platform, essential for collaboration and community building.

Many active and dedicated members of the community find themselves hindered by the current policy, which can inadvertently favor long-term inactive accounts over those actively contributing. I propose we discuss potential adjustments that could benefit all users, such as implementing a system to reclaim inactive usernames without compromising user privacy.

Let's discuss this constructively: What are your thoughts and experiences regarding this policy? How do you think GitHub could improve its approach to better support the community's growth and engagement?

Best
Michael

**Reply by ettaboyle** [reply]
Thank you @michaelfreund I appreciate your response and understanding.

As we've stated before: We are no longer accepting requests to release dormant usernames. We are not able to accommodate this request. 

This is not something myself or my team have any oversight on, and we are respectfully sharing a business decision that has been made.

If you hold a trademark to this name and believe there is a clear case of confusion between the account and your trademark, please review our [Trademark Policy](https://docs.github.com/articles/github-trademark-policy/) to determine if you should submit a trademark policy violation report.

Thank you!

**Reply by mrclrchtr** [reply]
Thank you for the answer @ettaboyle, even if it is not satisfying.

Is there a way to communicate one's dissatisfaction about this to a department that is responsible for it?

Context: We are a company that has been trying for some time to contact a person who is obviously inactive, who we would like to ask if he would be willing to hand the account name over.

**Reply by miguelpeniche** [reply]
Sorry I didn't know about tagging GitHub staff, it won't happen again. But still, no one responded what we ask, you guys keep telling us the same script. I think is a good idea to open the discussion, it will be good for GitHub and the community.

**Comment by 4po** [comment]
> Hello,
> 
> If I want my organization to be called _X_ and _X_ is occupied by an account created 5 years or more ago that has 0 activities since its creation, can I claim _X_ for my organization?
> 
> If so, how do I do it?
> 
> Thanks and sorry for my English.

qe

**Comment by nullbio** [comment]
This policy needs to be revised, especially in the age of AI automation where bots are squatting every English word combination they can think of. The username I want for my org is taken by someone who has never made  a single commit, for years... Pretty ridiculous.
